### PR TITLE
doc: Indicate -rpcauth option password hashing alg

### PR DESCRIPTION
**Pull request by dongcarl**
By indicating the password hashing algorithm, users of bitcoin distributions without the script in `share/rpcauth` and users who don't want to rely on said script can use alternative means to generate the password hash.

Question for reviewers: perhaps we should also indicate that it is specifically a HMAC-SHA-256 of the _**UTF-8**_ encoding of their password?

### CODE REVIEW

File: doc/man/bitcoind.1 [hunk 1]
```diff
@@ -530,7 +530,7 @@ option can be specified multiple times
 .HP
 \fB\-rpcauth=\fR<userpw>
 .IP
-Username and hashed password for JSON\-RPC connections. The field
+Username and HMAC-SHA-256 hashed password for JSON\-RPC connections. The field
```
Comment (MarcoFalke) [inline doc/man/bitcoind.1]: No need to modify these files., they are autogenerated.

Comment (dongcarl) [reply doc/man/bitcoind.1]: Oh? If they're autogenerated then why are they version controlled?

Comment (MarcoFalke) [reply doc/man/bitcoind.1]: https://github.com/bitcoin/bitcoin/pull/8608#issuecomment-242899755

**Comment by dongcarl** [comment]
Fixed.

**Comment by laanwj** [comment]
utACK

> Question for reviewers: perhaps we should also indicate that it is specifically a HMAC-SHA-256 of the UTF-8 encoding of their password?

Maybe in some more detailed spec document, but I don't think the option help is a place to go into such detail. It's consistent with the use of UTF-8 throughout the entire application. Mentioning it for the specific option might make people think "then where should I not use UTF-8?".

**Review by ryanofsky — APPROVED** [review]
utACK dcb70b152292a8df9fe834bacdf231ea848819b1. Could add `doc:` or `[doc]` prefix to PR description to indicate this a documentation-only change.

**Comment by fanquake** [comment]
utACK dcb70b1

**Pull request merged**